mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -222,134 +280,423 @@ public void testSimpleAttachment3() throws MessagingException, IOException {
+        expectedException.expectMessage("No value for pattern parameter was provided.");
+        mailet.init(mci);
     }
 
     @Test
-    public void testToAndFromAttributes() throws MessagingException,
-            IOException {
-        Mailet strip = new StripAttachment();
+    public void initShouldThrowOnWrongPattern() throws MessagingException {
+        Mailet mailet = new StripAttachment();
+
         FakeMailetConfig mci = FakeMailetConfig.builder()
                 .mailetName("Test")
-                .setProperty("attribute", "my.attribute")
+                .setProperty("directory", folderPath)
                 .setProperty("remove", "all")
-                .setProperty("notpattern", ".*\\.tmp.*")
+                .setProperty("pattern", ".****\\.tmp")
                 .build();
-        strip.init(mci);
 
-        Mailet recover = new RecoverAttachment();
-        FakeMailetConfig mci2 = FakeMailetConfig.builder()
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not compile regex [.****\\.tmp]");
+        mailet.init(mci);
+    }
+
+    @Test
+    public void initShouldThrowOnWrongNotPattern() throws MessagingException {
+        Mailet mailet = new StripAttachment();
+
+        FakeMailetConfig mci = FakeMailetConfig.builder()
                 .mailetName("Test")
-                .setProperty("attribute", "my.attribute")
+                .setProperty("directory", folderPath)
+                .setProperty("remove", "all")
+                .setProperty("notpattern", ".****\\.tmp")
                 .build();
-        recover.init(mci2);
 
-        Mailet onlyText = new OnlyText();
-        onlyText.init(FakeMailetConfig.builder()
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not compile regex [.****\\.tmp]");
+        mailet.init(mci);
+    }
+
+    @Test
+    public void initShouldSetRemoveAttachmentsToMatched() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
+
+        FakeMailetConfig mci = FakeMailetConfig.builder()
                 .mailetName("Test")
-                .build());
+                .setProperty("directory", folderPath)
+                .setProperty("remove", "matched")
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.removeAttachments).isEqualTo("matched");
+    }
 
+    @Test
+    public void serviceShouldThrowWhenUnretrievableMessage() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
+        
+        Mail mail = mock(Mail.class);
+        when(mail.getMessage())
+            .thenThrow(new MessagingException("Test exception"));
+
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not retrieve message from Mail object");
+        
+        mailet.service(mail);
```

Comment:
If you use service, please init your mailet

In reallity you mailet will be inited before receiving emails (or it's a complete new class of tests)